bytecomp/simplif.ml

```diff
@@ -754,7 +754,8 @@ let simplify_local_functions lam =
       -> false
   in
   let rec tail = function
-    | Llet (_str, _kind, id, Lfunction lf, cont) when enabled lf.attr ->
+    | Llet (_str, _kind, id, Lfunction lf, cont)
+      when Lambda.function_is_curried lf && enabled lf.attr ->
```

Comment:
Since that's the only use of  Lambda.function_is_curried that would remain after the other note is taken into account, this could just be:
````ocaml
 | Llet (_str, _kind, id, Lfunction ({kind=Curried} as lf), cont) when enabled lf.attr ->
````
(avoiding the need to define Lambda.function_is_curried)

Reply:
True, but I generally favour the "accessor" style here (it would ease any future attempt at adding more abstraction, for example).